src/npc.h

```diff
@@ -221,6 +227,7 @@ class Npc final : public Creature
 		std::map<std::string, std::string> m_parameters;
 
 		std::set<Player*> shopPlayerSet;
+		std::set<Player*> spectators;
```

Comment:
Because I couldn't find a reliable way to handle players both appearing, disappearing, logging in and out near the NPC.

If I replace all inserts with an increment on a counter and all erases with a decrement, it sometimes, but not always, gets double effect and the NPC gets to zero spectators when someone is nearby or -1 when someone goes away.

A set of pointers does not add so much overhead while keeping a natural O(1) check for duplicate entries.